### PR TITLE
Use unified flat filling function

### PR DESCRIPTION
**Pull request by JNechaevsky**
As discussed before, this should rationally simplify flat drawing/filling code.

### CODE REVIEW

**Comment by JNechaevsky** [comment]
@fabiangreffrath, have a look please before I start digging into other games. Few remarks:
* My initial idea of simple function for filling whole screen wasn't optional, so here is it's extended version for handling all possible cases. This approach should also make diff as small as possible.
* Is comment "use unified flat filling function" okay? Probably it is not needed in Crispy-specific functions? And the name `V_FillFlat` itself?
* M_DrawCrispnessBackground:
  * I have to un-const `crispness_background`, otherwise I'll need to make `const` the rest of `src`, which is a huge and unnecessary diff and not smart move.
  * `PU_STATIC` is potentially dangeours for flats handling! Possible case: `CRISPYBG` is used as common flat, which is handled by `PU_CACHE`, but in menu it's handled by `PU_STATIC`. This is a direct way to silent crash sooner or later, despite of such situation unlikely to ever happen.

**Comment by fabiangreffrath** [comment]
This is looking nice already!

**Comment by JNechaevsky** [comment]
Looks likes it's all done and ready to be reviewed.

File: src/doom/st_stuff.c [hunk 1]
```diff
@@ -407,25 +407,16 @@ void ST_refreshBackground(boolean force)
 	// so it appears to the left and right of the status bar in widescreen mode
 	if ((SCREENWIDTH >> crispy->hires) != ST_WIDTH)
 	{
-		int x, y;
+		int x;
```
Comment (fabiangreffrath) [inline src/doom/st_stuff.c:410]: Could you move the `x` into its own scope inside the `if (scaledviewwidth == SCREENWIDTH)` check?

File: src/heretic/r_draw.c [hunk 2]
```diff
@@ -556,7 +539,7 @@ void R_DrawTopBorder(void)
 {
     byte *src;
     pixel_t *dest;
-    int x, y;
+    int x;
```
Comment (fabiangreffrath) [inline src/heretic/r_draw.c:542]: Same herr, could you move the `x`, please?

**Review by fabiangreffrath — APPROVED** [review]
Looks good, thank you! If you could move the stray remaining `x` variables into their scopes, that'd be great.

**Comment by fabiangreffrath** [comment]
Thank you! 😽

**Comment by JNechaevsky** [comment]
Sure, was done in https://github.com/fabiangreffrath/crispy-doom/pull/1116/commits/71d9ab36b8ed77b0fffa29c70746f071de895cfc. I haven't thinked about it, but for sure it was a good idea.

Thanks for accepting! 🙇🏻‍♀️ This should be good for farther working on true color, or simply fore some longevity in case something will be changed in video code. Guess I can start working on true color support for Hexen now. Oh my, why... 😀

**Pull request merged**